models/challenge_model.go

```diff
@@ -1,15 +1,49 @@
 package models
 
 import (
+	"context"
+	"log"
+
 	"github.com/kamva/mgm/v3"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
 )
 
+type Location struct {
+	Type        string    `json:"type" bson:"type"`
+	Coordinates []float64 `json:"coordinates" bson:"coordinates" binding:"required"`
+}
+
 type Challenge struct {
 	mgm.DefaultModel `bson:",inline"`
-	Longitude        int    `bson:"longitude" json:"longitude"`
-	Latitude         int    `bson:"latitude" json:"latitude"`
-	Address          string `bson:"address" json:"address"`
-	TaskName         string `bson:"taskName" json:"taskName" binding:"required"`
-	Description      string `bson:"description" json:"description" binding:"required"`
-	Points           int    `bson:"points" json:"points" binding:"required"`
+	Location         Location `bson:"location" json:"location"`
+	Address          string   `bson:"address" json:"address"`
+	TaskName         string   `bson:"taskName" json:"taskName" binding:"required"`
+	Description      string   `bson:"description" json:"description" binding:"required"`
+	Points           int      `bson:"points" json:"points" binding:"required"`
+}
+
+func (model *Challenge) Saving() error {
+	if err := model.DefaultModel.Saving(); err != nil {
+		return err
+	}
+
+	if model.Location.Coordinates != nil {
+		// only need to worry about coordinates when creating challenges
```

Comment:
nit: Can we make comments start with a capital and end with a period.

Reply:
Also, this comment is a bit confusing since it's in the `Saving` function. I would say either remove it or change to something along the lines of `Set the location type to "Point" for when Challenges are created or updated`